### ISSUE TITLE
Nanite Cultivator output ignored on field (Bug report)

### ISSUE THREAD

**Issue by KevinGelking** [zymex22/Project-RimFactory-Revived#376]
When using the 'nanite' type cultivators in the center of a growth zone, using a 'growth zone puller' to transfer items, the nanite cultivator will fail to output the harvested items in the designated output tile and instead place them onto itself.
The tile in question is not blocked (it's just soil with crops growing on it).

The 'growth zone puller' cannot reach the item on top of the nanite cultivator as it's not part of the growth zone. For the same reason, leaving an empty tile (in case crops count as 'blocked tile') for the output is undesirable as it would not be part of the growth zone area required for the 'growth zone puller' to work.

**The expected behaviour would be:**
The Cultivator should output the items exclusively to the output tile. If blocked by another item, the cultivator should ideally have a behaviour option: 'place on itself' / 'place nearby' / 'pause' as placing the item on itself almost always requires pawn interaction.

Reproducibility: 10/10
**Steps to reproduce the behavior:**
1. Place nanite cultivator on soil
2. Generate a whole Growth Zone area around the cultivator
3. Place a Growth Zone Puller at the outer edge of the Growth Zone
4. Use dev mode to 'grow to maturity' on a few crops
5. Observe the items being dropped on top of the cultivator rather than in the output tile
6. Observe also that the Growth Zone Puller is unable to pull items from there (as expected)

See screenshot below with marked output tile and actual item drop.
<img width="322" alt="dllhost_U5TQGBNeXZ" src="https://user-images.githubusercontent.com/30538130/105223437-13120900-5b5c-11eb-8db4-43da6c524753.png">

**Modlist (from ModConfig):**
- brrainz.harmony
- ludeon.rimworld
- unlimitedhugs.hugslib
- odeum.wmbp
- oskarpotocki.vanillafactionsexpanded.core
- vanillaexpanded.vfecore
- vanillaexpanded.vfemedical
- vanillaexpanded.achievements
- vanillaexpanded.vwe
- vanillaexpanded.vappe
- vanillaexpanded.varme
- vanillaexpanded.vaecd
- vanillaexpanded.vaeas
- vanillaexpanded.vfesecurity
- vanillaexpanded.vaed
- vanillaexpanded.vaeed
- vanillaexpanded.vfeproduction
- vanillaexpanded.vhe
- vanillaexpanded.vaet
- vanillaexpanded.vaeis
- vanillaexpanded.vaebf
- vanillaexpanded.vweq
- vanillaexpanded.vaetr
- vanillaexpanded.vaets
- vanillaexpanded.vcef
- vanillaexpanded.vee
- vanillaexpanded.vfefarming
- vanillaexpanded.vaeau
- vanillaexpanded.vfeart
- vanillaexpanded.vwel
- vanillaexpanded.vtexe
- oskarpotocki.vanillafactionsexpanded.medievalmodule
- vanillaexpanded.vfespacer
- oskarpotocki.vanillafactionsexpanded.settlersmodule
- vanillaexpanded.vfepower
- vanillaexpanded.vfepropsanddecor
- vanillaexpanded.vcooke
- vanillaexpanded.vplantse
- vanillaexpanded.vcookestews
- vse.perrypersistent
- vse.empressevil
- oskarpotocki.vfe.insectoid
- vanillaexpanded.vcookesushi
- vanillaexpanded.vbrewe
- vanillaexpanded.vhe.beards
- vanillaexpanded.vbookse
- vanillaexpanded.vweg
- vanillaexpanded.vplantsesucculents
- oskarpotocki.vfe.vikings
- vse.freyafierce
- vse.igorinvader
- vanillaexpanded.vaetf
- vanillaexpanded.vbrewecandt
- vanillaexpanded.vanillatraitsexpanded
- oskarpotocki.vfe.mechanoid
- vanillaexpanded.vwehw
- oskarhpotocki141.vanillahanimalshexpandedhmhlivestock
- spdskatr.projectrimfactory
- kentington.saveourship2
- edb.preparecarefully
- oblitus.mylittleplanet
- radzerp.neorimnauts
- falconne.heatmap
- indeed.stackxxl
- fyarn.priorityclean
- codeoptimist.jobsofopportunity
- uuugggg.replacestuff
- orion.hospitality
- kittahkhan.grazeup
- jaxe.rimhud
- ykara.epoe
- automatic.bionicicons
- erdelf.minifyeverything
- ronco.dontstealmywalls
- fluffy.medicaltab
- petetimessix.simplesidearms
- fluffy.blueprints
- smuffle.harvestorganspostmortem
- automatic.recipeicons
- mlie.clothingsorter
- notfood.outfitted
- savestoragesettings.kv.rw
- bs.mbifvte
- mlie.roadsoftherim
- mlie.bestmix
- mlie.stockpilestacklimit

**Comment by zymex22** [comment]
im not sure why it output on top i need to test that.
but it will always output around it self an not around the zone =) 

Growzone puller can pickup if its dropped "in a grow zone"
Alternatively you can use Drone cultivators that can drop where ever you want them as they can do "hauling within its zone"

**Comment by joshuafcole** [comment]
I can second this issue. Anecdotally it seems to output on top of itself preferentially and then onto the output tile (or the output tile I chose just happens to be the second tile it chooses to fill _shrug_).

**Comment by Sn1p3rr3c0n** [comment]
The Cultivator will always pace the Items to the "best" cell in a small radius on and around the selected output cell.
Therefor if one where to place an IO Port on the Output cell the Described Scenario will never occur.

so generally i would say that this is expected behavior.

However I agree that the placement on top of the Cultivator is not Ideal.
we can disallow the placement on top of itself

---

Another option would be to switch the cultivator over to the Placement Queue system  _(just like the assemblers)_

**Comment by KevinGelking** [comment]
I/O ports are interesting, but they do consume quite a lot of power. Ideally for growth zones I'd like to be able to use the "growth zone puller" which can handle an area Spannung multiple individual cultivators so long they share the same zone.

My main point is though that I don't really see why the cultivator should in my case not place the items in the output tile, which is obviously not blocked by another item 🤔

My World is plagued by psychic rain, so I'm trying to keep my pawns from having to go out on the farms to collect items xD

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
